python/python/tests/test_async_client.py

```diff
@@ -1213,6 +1214,64 @@ async def test_persist(self, redis_client: TRedisClient):
         assert await redis_client.expire(key, 10)
         assert await redis_client.persist(key)
 
+    @pytest.mark.parametrize("cluster_mode", [True, False])
+    @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
+    async def test_geoadd(self, redis_client: TRedisClient):
+        key, key2 = get_random_string(10), get_random_string(10)
+        members_coordinates = {
+            "Palermo": Coordinate(13.361389, 38.115556),
+            "Catania": Coordinate(15.087269, 37.502669),
+        }
+        assert await redis_client.geoadd(key, members_coordinates) == 2
+        members_coordinates["Catania"].latitude = 39
+        assert (
+            await redis_client.geoadd(
+                key,
+                members_coordinates,
+                existing_options=ConditionalChange.ONLY_IF_DOES_NOT_EXIST,
+            )
+            == 0
+        )
+        assert (
+            await redis_client.geoadd(
+                key,
+                members_coordinates,
+                existing_options=ConditionalChange.ONLY_IF_EXISTS,
+            )
+            == 0
+        )
+        members_coordinates["Catania"].latitude = 40
+        members_coordinates.update({"Tel-Aviv": Coordinate(32.0853, 34.7818)})
+        assert (
+            await redis_client.geoadd(
+                key,
+                members_coordinates,
+                changed=True,
+            )
+            == 2
+        )
+
```

Comment:
You can query the key with `ZRANGE` to show/check its content
See example for [`geosearchstore`](https://redis.io/docs/latest/commands/geosearchstore/) at the very bottom

Reply:
this is not exactly true, its after storing the dist in a different key using geosearchstore